### PR TITLE
Ktx2 and meshopt integration

### PR DESCRIPTION
**Pull request by dinomut1**
## Summary

_A summary of changes being made in this PR_

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [x] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [x] If this PR is still a WIP, convert to a draft 
- [x] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

_References to pertaining issue(s)_

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`
try to load the "shopping-center" or "globe-theater" projects as locations.
List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

_Reviewers for this PR_

### CODE REVIEW

File: packages/engine/src/scene/functions/loaders/ModelFunctions.ts [hunk 1]
```diff
@@ -53,6 +55,9 @@ export const deserializeModel: ComponentDeserializeFunction = (
   addComponent(entity, ModelComponent, props)
 
   if (Engine.isEditor) getComponent(entity, EntityNodeComponent)?.components.push(SCENE_COMPONENT_MODEL)
+
+  initializeKTX2Loader(getGLTFLoader())
```
Comment (HexaField) [inline packages/engine/src/scene/functions/loaders/ModelFunctions.ts:59]: i dont think we want to be initializing a new instance of the loader every time a model loads, this should be initialized once in the SceneObjectSystem initializer, same as the draco loader

File: packages/engine/src/assets/functions/createGLTFLoader.ts [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
 import { DRACOLoader } from '../loaders/gltf/DRACOLoader'
 import { GLTFLoader } from '../loaders/gltf/GLTFLoader'
+import { KTX2Loader } from '../loaders/gltf/KTX2Loader'
+import { MeshoptDecoder } from '../loaders/gltf/meshopt_decoder.module'
 import { NodeDRACOLoader } from '../loaders/gltf/NodeDracoLoader'
 
+export const initializeKTX2Loader = (loader: GLTFLoader) => {
+  const ktxLoader: any = new KTX2Loader()
+  const THREE_PATH = `https://unpkg.com/three@0.${REVISION}.x`
+  ktxLoader.setTranscoderPath(`${THREE_PATH}/examples/js/libs/basis/`)
```
Comment (HexaField) [inline packages/engine/src/assets/functions/createGLTFLoader.ts:19]: we have the basis loader served from client public folder already, we don't want to be relying on external dependencies that we have no control over and potentially break

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/assets/functions/createGLTFLoader.ts (hunk 2 above)
Comment (dinomut1) [reply packages/engine/src/assets/functions/createGLTFLoader.ts]: The basis decoders are now being loaded from the public folders

File: packages/engine/src/scene/functions/loaders/ModelFunctions.ts (hunk 1 above)
Comment (dinomut1) [reply packages/engine/src/scene/functions/loaders/ModelFunctions.ts]: Added a null check to this such that it is only initialized once, and lazily as needed.

**Review by HexaField — APPROVED** [review]

**Pull request merged**